module_utils/certificate/providers/base.py

```diff
@@ -317,7 +320,7 @@ def email(self):
     @property
     def key_usage(self):
         """Return the Key Usage in the certificate."""
-        return self.cert_data.get("key_usage") or []
+        return set(self.cert_data.get("key_usage") or [])
```

Comment:
`self.cert_data.get("key_usage")` is already a `set` - and you can just call `set` with an empty constructor, you do not have to pass an empty `list` to the `set` constructor - I think this should be
`        return self.cert_data.get("key_usage") or set()`

Reply:
Yes. I forgot _get_key_usage will always return a set.